### PR TITLE
Allow config to load from within the jar file

### PR DESCRIPTION
**Pull request by ImDaBigBoss**

### CODE REVIEW

File: src/main/java/org/barrelmc/barrel/server/ProxyServer.java [hunk 1]
```diff
@@ -95,20 +93,38 @@ public ProxyServer(String dataPath) {
...
-        } catch (FileNotFoundException e) {
+        Path outputFile = this.dataPath.resolve("config.yml");
+
+        if (!Files.exists(outputFile)) {
             try {
-                InputStream inputStream = new FileInputStream("./src/main/resources/config.yml");
-                this.config = (new Yaml()).loadAs(inputStream, Config.class);
-                return true;
-            } catch (FileNotFoundException ignored) {
+                InputStream stream = Barrel.class.getClassLoader().getResourceAsStream("config.yml");
```
Comment (hYdos) [inline src/main/java/org/barrelmc/barrel/server/ProxyServer.java:100]: you haven't tested this (change it to `"/config.yml"`

File: src/main/java/org/barrelmc/barrel/server/ProxyServer.java [hunk 2]
```diff
@@ -74,7 +72,7 @@ public ProxyServer(String dataPath) {
         instance = this;
         this.dataPath = Paths.get(dataPath);
         if (!this.initConfig()) {
-            System.out.println("Config file not found! Terminating...");
+            System.out.println("Failed to read config file! Terminating...");
```
Comment (hYdos) [inline src/main/java/org/barrelmc/barrel/server/ProxyServer.java:75]: ???

**Review by hYdos — CHANGES_REQUESTED** [review]

**Comment by hYdos** [comment]
It would also be better if this pr just got closed and instead we wrote the file to the run directory instead of this

File: src/main/java/org/barrelmc/barrel/server/ProxyServer.java (hunk 1 above)
Comment (ImDaBigBoss) [reply src/main/java/org/barrelmc/barrel/server/ProxyServer.java:100]: I did test it

**Comment by ImDaBigBoss** [comment]
That's a terrible idea, it would mean that you need a file structure before being able to run this program. Why can't Barrel just make it for you? I'm going to let a maintainer of this project decide, I don't come and s*** on your code.

**Pull request closed without merging**